anthropic-java-bedrock/src/main/kotlin/com/anthropic/bedrock/backends/BedrockBackendAdapter.kt

```diff
@@ -0,0 +1,592 @@
+import software.amazon.awssdk.http.ContentStreamProvider
+import software.amazon.awssdk.http.SdkHttpMethod
+import software.amazon.awssdk.http.SdkHttpRequest
+import software.amazon.awssdk.http.auth.aws.signer.AwsV4HttpSigner
+import software.amazon.awssdk.http.auth.spi.signer.SignRequest
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain
+import software.amazon.eventstream.MessageDecoder
+
+/**
+ * An adapter for the Amazon Bedrock backend service that manages the AWS
+ * credentials required to access an Anthropic AI model on Bedrock and adapts
+ * requests and responses to Bedrock's requirements.
+ *
+ * Amazon Bedrock requires cryptographically-signed requests using credentials
+ * issued by AWS. These can be provided via system properties, environment
+ * variables, or other AWS facilities. They can be resolved automatically by
+ * the default AWS provider chains by calling [Builder.fromEnv]. Alternatively,
+ * the AWS credentials and region can be resolved independently and passed to
+ * [Builder.awsCredentials] and [Builder.region] should an alternative method
+ * of resolution be required.
+ */
+class BedrockBackendAdapter private constructor(
+    /**
+     * The AWS credentials required to access the Bedrock service.
+     */
+    @get:JvmName("awsCredentials") val awsCredentials: AwsCredentials,
+
+    /**
+     * The name of the AWS region hosting the Bedrock service.
+     */
+    @get:JvmName("region")  val region: Region,
+) : BackendAdapter {
+
+    /**
+     * A mapper for serializing and deserializing JSON data. For efficiency,
+     * this is created once and then reused for the life of this instance.
+     */
+    private val jsonMapper = jsonMapper()
+
+    companion object {
+        /**
+         * The Amazon Bedrock service name used for model inferences.
+         */
+        private const val SERVICE_NAME = "bedrock-runtime"
+
+        /**
+         * The Anthropic version for the Bedrock service.
+         */
+        private const val ANTHROPIC_VERSION = "bedrock-2023-05-31"
+
+        /**
+         * The name of the header that identifies the Anthropic beta versions.
+         * To specify multiple beta versions, there can be more than one header,
+         * or the value of a header can be a comma-separated list of beta
+         * versions.
+         */
+        // For HTTP/2 responses, the header names are expected to be lower case.
+        private const val HEADER_ANTHROPIC_BETA = "anthropic-beta"
+
+        /**
+         * The name of the header that identifies the content type of the data
+         * in the response.
+         */
+        private const val HEADER_CONTENT_TYPE = "content-type"
+
+        /**
+         * The name of the header that identifies the content type for the
+         * "payloads" of AWS _EventStream_ messages in streaming responses from
```

Comment:
What's up with the underscores around `EventStream`, here and below? Should it be "inline code" instead via backticks?